integration/test/Test/Brig.hs

```diff
@@ -18,3 +21,74 @@ testSearchContactForExternalUsers = do
 
   bindResponse (Public.searchContacts partner (owner %. "name")) $ \resp ->
     resp.status `shouldMatchInt` 403
+
+testCrudFederationRemotes :: HasCallStack => App ()
+testCrudFederationRemotes = do
+  let parseFedConns :: HasCallStack => Response -> App [Internal.FedConn]
+      parseFedConns resp = do
+        -- TODO: not idiomatic!  try `getJSON 200 resp %. "remotes" & asList & mapM asObjOrSomething`
+        -- Some ideas: There is asList to assert that a Value is actually a an array of Values. Then you can sort that to have a defined order.
+        fromJust . Aeson.decode . Aeson.encode . fromJust <$> ((`lookupField` "remotes") =<< getJSON 200 resp)
```

Comment:
Why is there a round trip through Aeson here?

Reply:
It was doing `ByteString -> Maybe a` by re-encoding the already parsed JSON. Using some other aeson functions allows us to get at the desired parse result more directly.

Reply:
Or don't parse at all! This is the way

Reply:
I've removed more of the FedConn specific code, and now it is comparing json values and leaning on typeclasses to align values.